client/src/app/app.component.ts

```diff
@@ -55,24 +56,8 @@ export class AppComponent implements OnInit {
         this.isLoading = false;
       }
     });
-  }
 
-  private setSsoMode() {
-    this.authenticationService.sso = true;
-    this.router.navigate([], {
-      queryParams: {
-        sso: null
-      }
-    });
-  }
-
-  showLogoutButton() {
-    this.router.events.subscribe(event => {
-      if (event instanceof NavigationStart && (event.url === 'login' || event.url === 'logout')) {
-        return false;
-      }
-      return true;
-    });
+    this.checkShowLogoutButton();
```

Comment:
Maybe just a minor thing: a few lines above we are already looking at ``router.events``. Maybe the code can be joined, e.g. in the following way:
```
this.router.events.subscribe(event => {
  if (event instanceof NavigationStart) {
    if (event.url === '/') {
      this.loadAllProjects();
    } else if (event.url === 'login' || event.url === 'logout') {
      this.showLogoutButton = false;
    }
    this.isLoading = false;
  }
});
```

BTW: I'm wondering, if ``this.isLoading = false;`` should not be skipped when calling ``this.loadAllProjects();``.

Reply:
Good point! And `this.isLoading` should be in the `else if` part, as `this.loadAllProjects()` sets this on its own.